lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -314,11 +333,22 @@ def _install_remote_rpms(base, filenames):
         base.package_install(pkg)
 
 
-def ensure(module, base, state, names):
+def ensure(module, base, state, names, autoremove):
     # Accumulate failures.  Package management modules install what they can
     # and fail with a message about what they can't.
     failures = []
     allow_erasing = False
+
+    # Autoremove is called alone
+    # Autoremove api was introduced at dnf 2.0.1-1
+    if not names:
+        names = []
+        if autoremove:
+            if dnf.__version__ >= u'2.0.1':
```

Comment:
version compares should be done differently (as, for instance, dnf-10.0 should satisfy the check).  You can use something like this:

``` python
from distutils.version import LooseVersion
[...]
if LooseVersion(dnf.__version__) >= LooseVersion('2.0.1'):
```